RestaurantMap/Scener/Home/HomeViewController.swift

```diff
@@ -24,13 +34,103 @@ class HomeViewController: UIViewController {
         let presenter = HomePresenter()
         presenter.view = self
         presenter.router = router
-        interactor?.presenter = presenter
+        interactor.presenter = presenter
         router?.navigationController = self.navigationController
     }
+    
+    private func setupMapView() {
+        self.mapView.delegate = self
+        do {
+            guard let styleURL = Bundle.main.url(forResource: "StyleGoogleMap", withExtension: "json")
+            else {
+                 print("Unable to find style.json")
+                return
+              }
+            self.mapView.mapStyle = try GMSMapStyle(contentsOfFileURL: styleURL)
+           } catch {
+               print("One or more of the map styles failed to load. \(error)")
+          }
+    }
+    
+    private func setupLocationManager() {
+        locationManager = CLLocationManager()
+        locationManager.desiredAccuracy = kCLLocationAccuracyBest
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.startUpdatingLocation()
+        locationManager.delegate = self
+    }
+    
+    private func showMarker(locationCoordinate: CLLocationCoordinate2D) {
```

Comment:
Separate this function to three functions
1. Show main marker
2. Clear map view
3. Fetch new restaurants with new location